includes/active-directory-develop-error-logging-introduction.md

```diff
@@ -8,19 +8,21 @@ ms.topic: include
 # Purpose:
 # Ingested by Microsoft identity platform articles in /articles/active-directory/develop/* that document error logging for the different platforms.
 ---
-The Microsoft Authentication Library (MSAL) apps generate log messages that can help diagnose issues. An app can configure logging with a few lines of code, and have custom control over the level of detail and whether or not personal and organizational data is logged. We recommend you create an MSAL logging callback and provide a way for users to submit logs when they have authentication issues.
+MSAL.NET takes advantage of the `IIdentityLogger` interface to provide logging messages to applications (MSAL.NET 4.45.0+). This comes with the benefit of enabling one logger implementation to be sharable between our partner SDks (Microsoft.Identity.Web, Microsoft.IdentityModel). In order to take advantage of this new API you will need to provide an implementation of the `IIdentityLogger` interface. This enables you to dynamically change the behavior of the logger without needing to rebuild your application. For example, you could configure the log level in the `IsEnabled` method to use a environment variable for greater flexibility during debugging.
```

Comment:
This page seems generic to all MSALs? Maybe we don't need to change it.

Reply:
Oh I see, the introduction file is shared amongst all MSALS. However, it is technically incorrect now since we have different logging levels.

Reply:
I may have to create a new introduction just for MSAL.NET

Reply:
And if you are using Id.Web you don't need to provide an implementation of IIdentityLogger.

Reply:
@trwalke I'll take a closer look at this PR today to see how best to approach this. It's possible this is a good candidate for implementing the "code-first" documentation approach and I'd like to help show how that's done.

I'll get back to you later today one way or the other (whether I get to it or not, but I should be able to).

Also, in your comment here, what is the "it" you're referring to:

> Oh I see, the introduction file is shared amongst all MSALS. However, **it is technically incorrect **now since we have different logging levels.

Is your new introduction incorrect (doubtful), or are you saying the existing intro text in the include file is incorrect? If it's the latter, can you clarify how it's incorrect? Just MSAL.NET, all MSALs, etc.

Thanks!

Reply:
@trwalke UPDATE: Not going to have time to take a deep dive on this. Will address the larger code-first paradigm separately.

For now, any changes in the include file must be applicable and be written to apply generically to all MSALs. This change, in particular, definitely needs to be backed out.